src/irmin-test/store.ml

```diff
@@ -417,6 +421,76 @@ module Make (S : S) = struct
+      with_contents repo (fun t -> P.Contents.add t "3") >>= fun kv3 ->
+      with_node repo (fun g -> Graph.v g [ ("x3", normal kv3) ]) >>= fun kt3 ->
+      with_contents repo (fun t -> P.Contents.add t "4") >>= fun kv4 ->
+      with_node repo (fun g -> Graph.v g [ ("x4", normal kv4) ]) >>= fun kt4 ->
+      with_contents repo (fun t -> P.Contents.add t "5") >>= fun kv5 ->
+      with_node repo (fun g -> Graph.v g [ ("x5", normal kv5) ]) >>= fun kt5 ->
+      with_contents repo (fun t -> P.Contents.add t "6") >>= fun kv6 ->
+      with_node repo (fun g -> Graph.v g [ ("x6", normal kv6) ]) >>= fun kt6 ->
+      with_contents repo (fun t -> P.Contents.add t "7") >>= fun kv7 ->
+      with_node repo (fun g -> Graph.v g [ ("x7", normal kv7) ]) >>= fun kt7 ->
+      with_contents repo (fun t -> P.Contents.add t "8") >>= fun kv8 ->
+      with_node repo (fun g -> Graph.v g [ ("x8", normal kv8) ]) >>= fun kt8 ->
+      (* kr0 <- kr1 <-- kr3 <-- kr5   and   kr7 <- kr8
+                      \       /
+                kr2 <-- kr4 <-- kr6 *)
+      let with_info n fn = with_commit repo (fun h -> fn h ~info:(info n)) in
+      with_info 0 (History.v ~node:kt0 ~parents:[]) >>= fun (kr0, _) ->
+      with_info 1 (History.v ~node:kt1 ~parents:[ kr0 ]) >>= fun (kr1, _) ->
+      with_info 2 (History.v ~node:kt2 ~parents:[]) >>= fun (kr2, _) ->
+      with_info 3 (History.v ~node:kt3 ~parents:[ kr1 ]) >>= fun (kr3, _) ->
+      with_info 4 (History.v ~node:kt4 ~parents:[ kr1; kr2 ])
+      >>= fun (kr4, _) ->
+      with_info 5 (History.v ~node:kt5 ~parents:[ kr3; kr4 ])
+      >>= fun (kr5, _) ->
+      with_info 6 (History.v ~node:kt6 ~parents:[ kr4 ]) >>= fun (kr6, _) ->
+      with_info 7 (History.v ~node:kt7 ~parents:[]) >>= fun (kr7, _) ->
+      with_info 8 (History.v ~node:kt8 ~parents:[ kr7 ]) >>= fun (kr8, _) ->
+      History.closure h ~min:[ kr1 ] ~max:[ kr5 ] >>= closure_error
+      >>= fun krs ->
+      check_keys "g1" [ kr1; kr2; kr3; kr4; kr5 ] krs;
+      History.closure h ~min:[] ~max:[ kr5 ] >>= closure_error >>= fun krs ->
+      check_keys "g2" [ kr0; kr1; kr2; kr3; kr4; kr5 ] krs;
+      (* disconnected min and max, connected closure *)
+      History.closure h ~min:[ kr1; kr2 ] ~max:[ kr5; kr6 ] >>= closure_error
+      >>= fun krs ->
+      check_keys "g3" [ kr1; kr2; kr3; kr4; kr5; kr6 ] krs;
+      (* disconnected min and max, disconnected closure *)
+      History.closure h ~min:[ kr1; kr7 ] ~max:[ kr4; kr8 ] >>= closure_error
+      >>= fun krs ->
+      check_keys "g4" [ kr1; kr2; kr4; kr7; kr8 ] krs;
+      (* empty max raises error *)
+      (History.closure h ~min:[ kr7 ] ~max:[] >>= function
+       | Error `Empty_max -> Lwt.return_unit
+       | _ -> Alcotest.fail "Expected `Empty_max error")
+      >>= fun () ->
+      (* min node not connected raises error *)
+      (History.closure h ~min:[ kr7 ] ~max:[ kr6 ] >>= function
+       | Error (`Disconnected h) ->
+           check_key "kr7" h kr7;
+           Lwt.return_unit
```

Comment:
Could get rid of this line with a `>|=` rather than `>>=` (and similarly in the previous case).